src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -544,12 +544,31 @@ where
             let pkg = Box::new(source).download_now(pkgid, config)?;
             Ok(pkg)
         }
-        None => bail!(
-            "could not find `{}` in {} with version `{}`",
-            dep.package_name(),
-            source.source_id(),
-            dep.version_req(),
-        ),
+        None => {
+            let version: String = dep.version_req().to_string();
+            let pkg_id;
+            if dep.version_req().is_exact() {
+                pkg_id = PackageId::new(dep.package_name(), &version[1..], source.source_id());
+            } else {
+                pkg_id = PackageId::new(dep.package_name(), &version[..], source.source_id());
```

Comment:
Since this line is almost guaranteed to fail, I think it is a little misleading to try to create the PackageId here. Maybe be a little more explicit here?  For example, something like the following:

```rust
let is_yanked = if dep.version_req().is_exact() {
    let version: String = dep.version_req().to_string();
    PackageId::new(dep.package_name(), &version[1..], source.source_id())
        .map_or(false, |pkg_id| source.is_yanked(pkg_id).unwrap_or(false))
} else {
    false
};
```

Reply:
okay got it, like if we don't have exact version then control will always go to the else part. Let me update the PR. Thanks again for explaining your point with the code snippet, which really helps me to get your point.